### PR TITLE
implemented method to hash data

### PR DESCRIPTION
**Pull request by Awambeng**

### CODE REVIEW

**Comment by Awambeng** [comment]
done already

**Comment by Awambeng** [comment]
@stephane-segning we completed ticket [Implement a logic to hash and validate strings #14](https://github.com/ADORSYS-GIS/e2e-banking-app/issues/14) you can now review

File: power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/utils/DataSecurityService.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public String hashData(String data) {
+        return passwordEncoder.encode(data);
+    }
+
+    public boolean compareHashPassword(String pin, String phoneNumber) {
+        User user = userRepository.findById(phoneNumber).get();
+        String hashPin = user.getPin();
+        boolean result = passwordEncoder.matches(pin, hashPin);
+        return result;
+    }
```
Comment (stephane-segning) [inline power-pay-backend/src/main/java/com/adorsys/gis/powerpay/powerpaybackend/utils/DataSecurityService.java:25]: This should not have something to do with users. Just plain simple hash comparison with the real value. E.g `compareHashWithPassword("dlskdksd", "real-passord")`.

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good work @Awambeng ! You shouldn't make the service from the user repo btw. See these points

**Comment by Awambeng** [comment]
@stephane-segning work done as requested you can now review

**Comment by stephane-segning** [comment]
Your code does not compile. Please fix that @Awambeng @stephane-segning @forkimenjeckayang @chendiblessing

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Something's wrong in your implementation. Please refer to the official documentation of spring boot passwordencoder to validate your implementation of the check password. What you did cannot work.

**Comment by chendiblessing** [comment]
@stephane-segning Did some changes on the test compare hash function please review it

**Review by stephane-segning — APPROVED** [review]
Well done! Thank you @Awambeng @stephane-segning @chendiblessing @forkimenjeckayang

**Pull request merged**